src/test/java/com/reminiscence/reminiscence/userAccount/testAccountController.java

```diff
@@ -0,0 +1,128 @@
+
+    @Autowired
+    AccountRepo accountRepo;
+
+    @Autowired
+    PasswordEncoder encoder;
+
+    @Before
+    public void setUp() {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(webApplicationContext)
+                .apply(SecurityMockMvcConfigurers.springSecurity())
+                .build();
+    }
+
+    @Test
+    public void testDefaultConstructor() {
+        UserAccount user = new UserAccount();
+        String actualUsername = user.getUsername();
+        String actualPassword = user.getPassword();
+        String actualJoinDate = user.getJoinDate();
+
+        assertNull("Username had an unexpected value", actualUsername);
+        assertNull("Password had an unexpected value", actualPassword);
+        assertNull("Joindate had an unexpected value", actualJoinDate);
+    }
+
+    @Test
+    public void testLoadedConstructor() {
+        UserAccount user = new UserAccount("testUser", "testPassword", this.encoder, "2019-05-14");
+        String actualUsername = user.getUsername();
+        String actualPassword = user.getPassword();
+        String actualJoinDate = user.getJoinDate();
+
+        assertEquals("Username had an unexpected value", "testUser", actualUsername);
+        assertNotEquals("Password had an unexpected value", "testPassword", actualPassword);
+        assertEquals("Joindate had an unexpected value", "2019-05-14", actualJoinDate);
+    }
+
+    @Test
+    public void testMVCAccountCreation() throws Exception {
+        String username = "testy";
+        String password = "1234";
+        String date = "2019-05-14";
+
+        mockMvc.perform(MockMvcRequestBuilders.post("/signup")
+                .param("username", username)
+                .param("password", password)
+                .param("date", date))
+                .andExpect(status().is3xxRedirection())
+                .andExpect(redirectedUrl("/home"));
+    }
+
+    @Test
+    public void testAccountCreationUsernameTaken() throws Exception {
+        String username = "test";
+        String password = "1234";
+        String date = "2019-05-14";
+
+        mockMvc.perform(MockMvcRequestBuilders.post("/signup")
+                .param("username", username)
+                .param("password", password)
+                .param("date", date))
+                .andExpect(status().is3xxRedirection())
+                .andExpect(redirectedUrl("/home"));
+
+
+        String username2 = "test";
+        String password2 = "1234";
+        String date2 = "2019-05-14";
+        try {
+            mockMvc.perform(MockMvcRequestBuilders.post("/signup")
+                    .param("username", username2)
+                    .param("password", password2)
+                    .param("date", date2))
+                    .andExpect(status().isInternalServerError())
+                    .andExpect(redirectedUrl("/home"));
+        } catch (Exception E) {
+            System.out.println("Exception successfully thrown");

```

Comment:
This test won't fail!  (_any_ exception will be caught, and if the perform succeeds there is no fail() after it)

Instead:  check the specific type of exception.  (And add that fail())